test/modules/model/TinyLlamaWithFusedAttention/model.py

```diff
@@ -0,0 +1,87 @@
+import torch
+
+from tico.serialize.operators.adapters.llama_rmsnorm import patched_llama_rmsnorm
+from tico.serialize.operators.adapters.onert.llama_attention import (
+    llama_attention_forward_adapter,
+)
+from tico.utils.pytree_utils import register_dynamic_cache
+from tico.utils.record_input import RecordingInput
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from transformers.models.llama.modeling_llama import LlamaAttention
+
+from test.modules.base import TestModuleBase
+from test.utils import tag
+
+
+@tag.use_onert
+class TinyLlamaWithFusedAttention(TestModuleBase):
+    def __init__(self):
+        super().__init__()
+        self.model_name = "Maykeye/TinyLLama-v0"
+        self._call_count = 0
+        self.original_model = AutoModelForCausalLM.from_pretrained(
+            self.model_name
+        ).eval()
+        self.fused_model = AutoModelForCausalLM.from_pretrained(self.model_name).eval()
+        for layer in self.fused_model.model.layers:
+            layer.self_attn.forward = llama_attention_forward_adapter.__get__(
+                layer.self_attn
+            )
+        self.rtol = 1e-4
+        self.atol = 1e-4
+
+    def forward(self, *args, **kwargs):
+        self._call_count += 1
+
+        if self._call_count == 2:
+            return self.fused_model(*args, **kwargs)
+        else:
```

Comment:
TICO test framework get expected values using original model, then convert the model. For 1st run, it should not use adapter.

I tried not to use `call_count`, but to use `inspect` and get the caller's name. However, it is not allowed during torch tracing.

Reply:
I don't think it is a trick at all. `circle_custom.attention` is a virtual operator for fusing, not for computing. I found no reason to put the actual implementation here. 

The call_count related code can be eliminated if the TICO test framework is improved by isolating infering model and converting model. I see it already had problem:

https://github.com/Samsung/TICO/blob/b941ab29f3c4eacc622a9ef24261ec7ba6e76ef9/test/pt2_to_circle_test/builder.py#L139-L147